fs/base.py

```diff
@@ -318,6 +318,8 @@ def copydir(self, src_path, dst_path, create=False):
         with self._lock:
             if not create and not self.exists(dst_path):
                 raise errors.ResourceNotFound(dst_path)
+            if not self.getinfo(src_path).is_dir:
```

Comment:
Calling `getinfo` will raise a `ResourceNotFound` as expected, whereas if you were to use `isdir` you would need one call to check for the existence of a resource, and a second call to check that the resource is a directory.

Since the base implementation uses `getinfo` as a base for `isdir`, `isfile` and `exist`, it will result in only one call to `getinfo` to do as such but two calls otherwise.